components/tidb_query/src/codec/collation/utf8mb4.rs

```diff
@@ -400,24 +404,27 @@ impl Collator for CollatorUtf8Mb4Bin {
     type Charset = CharsetUtf8mb4;
 
     #[inline]
-    fn write_sort_key<W: BufferWriter>(
-        bstr: &[u8],
-        writer: &mut W,
-    ) -> Result<usize, DecodeOrWriteError> {
-        str::from_utf8(bstr).map_err(DecodeError::from)?;
+    fn write_sort_key<W: BufferWriter>(bstr: &[u8], writer: &mut W) -> Result<usize> {
+        str::from_utf8(bstr)?;
         writer.write_bytes(bstr)?;
         Ok(bstr.len())
     }
 
     #[inline]
-    fn sort_compare(a: &[u8], b: &[u8]) -> Result<Ordering, DecodeError> {
+    fn validate(bstr: &[u8]) -> Result<()> {
+        str::from_utf8(bstr)?;
```

Comment:
We already have validate() function, why do we need to validate again in `write_sort_key` and `sort_compare`?

Reply:
In fact, validate is not yet used... and I may remove it in later pr, if it's useless in fashhashaggr.

Reply:
Do we have similar validation for UTF8 strings before(I mean, before collations)?